### PR TITLE
feat(chip): add ionic theme and visual states

### PR DESCRIPTION
**Pull request by BenOsodrac**
Issue number: ROU-4840

---------

## What is the new behavior?

- Added a new Ionic Theme partial, with no dependencies to current `ios/md` architecture. This is to avoid bleeding of the current Design System into the work we are doing for the new Ionic Design System and Theme. For that reason I duplicated part of the basic styles for the Chip (display, box-sizing, etc) and created provisional scss variables, that should be replaced later on with the correct global ones from the new Design System.

- Added styles for info chip type states (default, hover, focus and disabled)

- No new tests were done on this scope, as there're still other Chip tasks being developed that would override any baselines defined now for this theme. Once the initial basic work for the Chip is complete, tests for the Ionic Theme should be implemented, under the task ROU-4837.

- A new page was created - `chip/tests/theme-ionic` - to check styles for current implementation. This will also be used and improved in future Chip Ionic Theme tasks.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/chip/chip.ionic.scss [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+:host(.chip-disabled) {
+  opacity: 0.6;
+  pointer-events: none;
+}
+
+// Chip: Focus
+// ---------------------------------------------
+
+:host(:focus) {
```
Comment (sean-perkins) [inline core/src/components/chip/chip.ionic.scss:69]: Focus management and styling is handled a little differently in Ionic Framework that traditional web experiences. 

We use the `.ion-focused` selector to target elements that are focusable. 

To make an element or component participate in this selector, you will need to add the `ion-focusable` CSS class to the element.

For example:
```diff
<Host
  aria-disabled={this.disabled ? 'true' : null}
  class={createColorClasses(this.color, {
    [theme]: true,
    'chip-outline': this.outline,
    'chip-disabled': this.disabled,
    'ion-activatable': true,
+   'ion-focusable': true
  })}
```

File: core/src/components/chip/chip.ionic.scss [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  color: var(--color);
+
+  font-family: $ionic-font-family;
+  font-size: $ionic-font-size-s;
+
+  cursor: pointer;
+
+  overflow: hidden;
+
+  vertical-align: middle;
```
Comment (sean-perkins) [inline core/src/components/chip/chip.ionic.scss:54]: I'm unsure if this style has an effect on the visual experience with the `align-items: center;` style above. I think we can remove it.

**Review by sean-perkins — CHANGES_REQUESTED** [review]

File: core/src/components/chip/chip.ionic.scss (hunk 2 above)
Comment (BenOsodrac) [reply core/src/components/chip/chip.ionic.scss:54]: Yeah, you probably right, I mindless copied this one from the existing chip.scss. Will remove it 👍

File: core/src/components/chip/chip.ionic.scss (hunk 1 above)
Comment (BenOsodrac) [reply core/src/components/chip/chip.ionic.scss:69]: Hum ok I wasn't sure about that actually, I thought about adding that but I saw that on current chip.scss we had this:

```
// Chip: Focus
// ---------------------------------------------

:host(:focus) {
  outline: none;
}

:host(:focus) {
  --background: #{rgba($text-color-rgb, 0.16)};
}
```

Does that mean is it also wrong there?
What exactly does that `ion-focusable` class behind the scenes?

Comment (BenOsodrac) [reply core/src/components/chip/chip.ionic.scss:68]: I already found the documentation explaining it 👍

Comment (sean-perkins) [reply core/src/components/chip/chip.ionic.scss:68]: Spoke with Liam to double check my understanding as well. The existing `chip.scss` implementation is wrong/old/outdated. 

`.ion-focused` is the correct implementation 👍

Comment (BenOsodrac) [reply core/src/components/chip/chip.ionic.scss]: Got it, thanks, already committed the code with `.ion-focused`!

**Comment by sean-perkins** [comment]
Looks like we just need to run `npm run build` and push up those changes.

**Pull request merged**